### ISSUE TITLE
Unable to use svg in Summer '17 org

### ISSUE THREAD

**Issue by aravindxz** [salesforce-ux/design-system#464]
> (REQUIRED) Include one or more screenshots if applicable, as well as a Codepen with the
reduced test case.
![svg issue](https://cloud.githubusercontent.com/assets/7611254/26678258/41bc48ba-46ed-11e7-9fd6-22f74457939b.png)

> Describe the issue. Is it a bug or a feature request (new component, new icon, new CSS class)?

In the official SLDS, website, many examples has used the `<svg>` tag (e.g. https://www.lightningdesignsystem.com/components/alert/ ).
When I try to save it, I get the above FIELD_INTEGRITY_EXCEPTION.

> Is this issue related to a specific component, variant, and/or state? If so, please detail which.

When trying to use `<svg>` tag.

> Are any specific browsers impacted by this bug?

Cannot even save the component in developer console!

> Which version of the Salesforce Lightning Design System are you using?

2.3.1, Summer '17 org

> What steps and/or code are needed to reproduce this issue?

Simple. Just copy the code provided under the "Developer Guidelines " in https://www.lightningdesignsystem.com/components/alert/, and try to save, you'll get the error.

> What did you expect to happen?

Expected to get the output as shown in the official SLDS website.

> What actually happened?

Cannot even save the component. 

Also, in the SLDS website, it's mentioned, from Spring '17 the svgIcon has been deprecated (Here's the link to that information: https://www.lightningdesignsystem.com/resources/lightning-svg-icon-component-helper/). What to do in this scenario?
Or is it that you should have the svgicon component for your code snippet provided in  "Developer Guidelines" to work?

**Comment by JohnHeadIV** [comment]
I've made other comments/issue about how the SLDS guidance is not copy/paste in to Lightning components (the most logical use case for SLDS). 
Have you tried using lightning:icon in place? 
We've continued to Use the custom SVG component with no problem (we like it because we can use both SLDS icon in it as well as our own internal set). I would add that lightning:icon throws a large numbers of warning in Summer 17 in LE. Our biggest issue is that extends="force:slds" is required for the lightning:icon to work, however force:slds is a application level attribute and not a component property which makes any component that utilizes lightning:icon more limited. (ie we found issues with Lightning for VF in previous releases).  
My vote is the SLDS needs to be either more tightly coordinated with its companion components or better separated. Following the SLDS guidance can cause issues when trying to actually match functionality/form within lightning components or SFDC in general.  End rant.

**Comment by aravindxz** [comment]
@JohnHeadIV I tried using lightning:icon and the icons were displayed. But I had to edit it's style(to set the icon colour) manually in the component's css file by calling something like 
`.THIS .lightningIconsClassName svg{fill: white;}` .

And yes, we're also using svg component, but as I have raised the case above, now in the site, it's being displayed as **deprecated**!! So, I'm not sure how to use these svg icon? If they've given such a code in their **"Developer Guidelines "**, then logically it should work, but unfortunately that is not the case here.
(I'm even facing scroll issues(in Salesforce1 app) for which I've raised an issue separately.)

And yes, you're right and even I totally agree with you regarding your take on SLDS.
